src/arrayMethodJoin.js

```diff
@@ -4,8 +4,20 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let result = '';
+
+    for (let i = 0; i < this.length; i++) {
+      if ((this[i] !== null) && (this[i] !== undefined)) {
+        result += this[i];
+      };
+
+      if (i !== this.length - 1) {
+        result += String(separator);
```

Comment:
no need to convert it to String, the separator that is added to the string (result) will be converted to the string automatically